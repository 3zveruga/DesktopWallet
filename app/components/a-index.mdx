---
name: Home
route: /
---

# UI Components

Use the sidebar menu to browse the styleguide of React components available for the Zepio application.
