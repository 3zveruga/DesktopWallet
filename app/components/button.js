// @flow

import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable max-len */
// $FlowFixMe
import { darken } from 'polished';

const DefaultButton = styled.button`
  padding: 10px 30px;
  font-family: ${// $FlowFixMe
  props => props.theme.fontFamily};
  font-weight: ${// $FlowFixMe
  props => props.theme.fontWeight.bold};
  font-size: ${// $FlowFixMe
  props => `${props.theme.fontSize.text}em`};
  cursor: pointer;
  outline: none;
  min-width: 100px;
  border-radius: 100px;
  transition: background-color 0.1s ease-in-out;
  width: 100%;
`;

const Primary = styled(DefaultButton)`
  background-color: ${props => props.theme.colors.primary};
  color: ${props => props.theme.colors.secondary};
  border: none;

  &:hover {
    background-color: ${props => darken(0.1, props.theme.colors.primary(props))};
  }

  &:disabled {
    background-color: #3e3c42;
    cursor: not-allowed;
    opacity: 0.8;
  }
`;

const Secondary = styled(DefaultButton)`
  background-color: transparent;
  color: ${props => props.theme.colors.secondary};
  border: 2px solid #3e3c42;

  &:hover {
    border-color: ${props => props.theme.colors.primary};
  }

  &:disabled {
    background-color: Transparent;
    cursor: not-allowed;
    color: #3e3c42;

    &:hover {
      border-color: #3e3c42;
    }
  }
`;

type Props = {
  label: string,
  onClick?: () => void,
  to?: string,
  variant?: 'primary' | 'secondary',
  disabled?: boolean,
  className?: string,
};

export const Button = ({
  onClick,
  label,
  to,
  variant,
  disabled,
  className,
}: Props) => {
  if (to && onClick) throw new Error('Should define either "to" or "onClick"');

  const component = variant === 'primary' ? (
    <Primary onClick={onClick} disabled={disabled} className={className}>
      {label}
    </Primary>
  ) : (
    <Secondary onClick={onClick} disabled={disabled} className={className}>
      {label}
    </Secondary>
  );

  return to ? <Link to={String(to)}>{component}</Link> : component;
};

Button.defaultProps = {
  to: null,
  variant: 'primary',
  onClick: null,
  disabled: false,
  className: '',
};
