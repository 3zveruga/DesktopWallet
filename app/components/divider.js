// @flow
import styled from 'styled-components';

export const Divider = styled.div`
  width: 100%;
  height: 1px;
  background-color: ${props => props.color || props.theme.colors.text};
  opacity: ${props => props.opacity || 1};
  margin-bottom: ${props => props.marginBottom || 0};
  margin-top: ${props => props.marginTop || 0};
`;
