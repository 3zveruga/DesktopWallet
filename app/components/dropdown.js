// @flow
import React, { type Node, Component } from 'react';
import styled from 'styled-components';
/* eslint-disable import/no-extraneous-dependencies */
// $FlowFixMe
import { darken } from 'polished';
import Popover from 'react-popover';
import ClickOutside from 'react-click-outside';

import { TextComponent } from './text';

/* eslint-disable max-len  */
const MenuWrapper = styled.div`
  background-image: ${props => `linear-gradient(to right, ${darken(
    0.05,
    props.theme.colors.activeItem,
  )}, ${props.theme.colors.activeItem})`};
  padding: 10px 0;
  border-radius: 10px;
  margin-left: -10px;
`;

const MenuItem = styled.button`
  outline: none;
  background-color: transparent;
  border: none;
  border-bottom-style: solid;
  border-bottom-color: ${props => props.theme.colors.text};
  border-bottom-width: 1px;
  padding: 15px 10px;
  cursor: pointer;
  width: 100%;
  text-align: left;

  &:hover {
    opacity: 0.9;
  }

  &:disabled {
    cursor: default;

    &:hover {
      opacity: 1;
    }
  }
`;

const PopoverWithStyle = styled(Popover)`
  & > .Popover-tip {
    fill: ${props => props.theme.colors.activeItem};
  }
`;

type Props = {
  renderTrigger: (toggleVisibility: () => void, isOpen: boolean) => Node,
  options: Array<{ label: string, onClick: () => void }>,
  label?: string | null,
};

type State = {
  isOpen: boolean,
};

export class DropdownComponent extends Component<Props, State> {
  state = {
    isOpen: false,
  };

  static defaultProps = {
    label: null,
  };

  render() {
    const { isOpen } = this.state;
    const { label, options, renderTrigger } = this.props;

    return (
      <PopoverWithStyle
        isOpen={isOpen}
        preferPlace='below'
        enterExitTransitionDurationMs={0}
        body={[
          <ClickOutside
            onClickOutside={() => this.setState(() => ({ isOpen: false }))}
          >
            <MenuWrapper>
              {label && (
                <MenuItem disabled>
                  <TextComponent value={label} isBold />
                </MenuItem>
              )}
              {options.map(({ label: optionLabel, onClick }) => (
                <MenuItem onClick={onClick}>
                  <TextComponent value={optionLabel} />
                </MenuItem>
              ))}
            </MenuWrapper>
          </ClickOutside>,
        ]}
        tipSize={7}
      >
        {renderTrigger(
          () => this.setState(state => ({ isOpen: !state.isOpen })),
          isOpen,
        )}
      </PopoverWithStyle>
    );
  }
}
