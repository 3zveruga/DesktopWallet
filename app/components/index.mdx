---
name: Home
route: /
---

# UI Components

Use the sidebar menu to browse the styleguide
