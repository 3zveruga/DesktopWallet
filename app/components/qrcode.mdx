---
name: QRCode
---

import { Playground, PropsTable } from 'docz'

import { QRCode } from './qrcode.js'

# QRCode

## Properties

<PropsTable of={QRCode} />

## Basic usage

<Playground>
  <QRCode value='https://z.cash.foundation' />
</Playground>

## Custom size

<Playground>
  <QRCode
    value='https://z.cash.foundation'
    size={500}
  />
</Playground>
