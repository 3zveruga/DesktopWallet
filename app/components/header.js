// @flow

import React from 'react';
import styled from 'styled-components';

import { ZcashLogo } from './zcash-logo';
import { TextComponent } from './text';
import { Divider } from './divider';
import { RowComponent } from './row';
import { StatusPill } from './status-pill';

import { withSyncStatus } from '../../services/sync-status';

const Wrapper = styled.div`
  height: ${(props: PropsWithTheme<>) => props.theme.headerHeight};
  width: 100vw;
  display: flex;
  flex-direction: row;
  background-color: ${(props: PropsWithTheme<>) => props.theme.colors.background};
`;

const LogoWrapper = styled.div`
  height: ${(props: PropsWithTheme<>) => props.theme.headerHeight};
  width: ${(props: PropsWithTheme<>) => props.theme.sidebarWidth};
  background-image: linear-gradient(
    to right,
    ${(props: PropsWithTheme<>) => props.theme.colors.sidebarLogoGradientBegin},
    ${(props: PropsWithTheme<>) => props.theme.colors.sidebarLogoGradientEnd}
  );
  margin-bottom: 20px;
`;

const TitleWrapper = styled.div`
  width: ${(props: PropsWithTheme<>) => `calc(100% - ${props.theme.sidebarWidth})`};
  height: ${(props: PropsWithTheme<>) => props.theme.headerHeight};
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
  padding-top: 10px;
  padding-left: ${(props: PropsWithTheme<>) => props.theme.layoutPaddingLeft};
  padding-right: ${(props: PropsWithTheme<>) => props.theme.layoutPaddingRight};
`;

const TitleRow = styled(RowComponent)`
  align-items: center;
  justify-content: space-between;
  width: 100%;
`;

const Title = styled(TextComponent)`
  font-size: ${(props: PropsWithTheme<>) => `${props.theme.fontSize.large}em`};
  margin-top: 10px;
  margin-bottom: 10px;
  text-transform: capitalize;
  letter-spacing: 0.25px;
  font-weight: ${(props: PropsWithTheme<>) => String(props.theme.fontWeight.bold)};
`;

type Props = {
  title: string,
};

const Status = withSyncStatus(StatusPill);

export const HeaderComponent = ({ title }: Props) => (
  <Wrapper id='header'>
    <LogoWrapper>
      <ZcashLogo />
    </LogoWrapper>
    <TitleWrapper>
      <TitleRow alignItems='center' justifyContent='space-around'>
        <Title value={title} />
        <Status type='syncing' progress={0} />
      </TitleRow>
      <Divider opacity={0.1} />
    </TitleWrapper>
  </Wrapper>
);
