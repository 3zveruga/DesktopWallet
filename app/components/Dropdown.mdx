---
name: DropDown
---

import { Playground, PropsTable } from 'docz'

import { Dropdown } from './dropdown.js'
import { DoczWrapper } from '../theme.js'

# DropDown

<PropsTable of={Dropdown} />

## Basic usage

<Playground>
  <DoczWrapper>
    {() => (
      <div style={{ height: '500px' }}>
        <Dropdown
          options={[
            { label: 'asbh1yeasbdh23848asdasd', onClick: console.log },
            { label: 'urtyruhjr374hbfdjdhuh', onClick: console.log },
          ]}
          renderTrigger={toggleVisibility => <button onClick={toggleVisibility}>Opa</button>}
        />
      </div>
    )}
  </DoczWrapper>
</Playground>
