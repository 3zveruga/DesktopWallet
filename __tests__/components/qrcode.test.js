// @flow

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import { QRCode } from '../../app/components/qrcode';

afterEach(cleanup);

describe('<QRCode />', () => {
  test('should render qrcode component correctly', () => {
    const { queryByTestId } = render(
      <QRCode value='https://z.cash.foundation' />,
    );

    expect(queryByTestId('QRCode')).toBeInTheDocument();
  });
});
