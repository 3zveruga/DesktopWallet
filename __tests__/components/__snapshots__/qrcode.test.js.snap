// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QRCode /> render() should render qrcode component correctly 1`] = `
<div>
  <canvas
    height="128"
    style="height: 128px; width: 128px;"
    width="128"
  />
</div>
`;
